### ISSUE TITLE
Support for Custom Azure Monitor Workspace Id for AKS

### ISSUE THREAD

**Issue by EppO** [hashicorp/terraform-provider-azurerm#20702]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently, there is no way to bind an existing Azure Monitor Workspace Id to an AKS cluster. The exisiting implementation (#19530 ) relies on a default Azure Monitor Workspace. This feature request is the first step to support custom Azure Monitor Workspace. Once #18809 is implemented, we could pass Azure Monitor Workspace resource or datasource to that new attribute. 

### New or Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster" "example" {
  name                = "example-aks1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  dns_prefix          = "exampleaks1"

  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_D2_v2"
  }

  monitor_metrics {
    azure_monitor_workspace_id = "/subscriptions/<subscriptionId>/resourcegroups/<resourceGroupName>/providers/microsoft.monitor/accounts/<azureMonitorWorkspaceName>"
  }

  identity {
    type = "SystemAssigned"
  }

  tags = {
    Environment = "Production"
  }
}
```

### References

- https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/prometheus-metrics-enable?tabs=cli#install-metrics-addon

```
az aks update --enable-azuremonitormetrics -n <cluster-name> -g <cluster-resource-group> --azure-monitor-workspace-resource-id <workspace-name-resource-id>
```

**Comment by cpressland** [comment]
@catriona-m I think this issue got closed accidentally, this functionality is still missing and the linked pull request doesn’t touch any Kubernetes components that I can see.

**Comment by EppO** [comment]
@cpressland: you're right, #21598 implemented #18809, not this one

**Comment by aristosvo** [comment]
@EppO @cpressland Although this feature is marketed as being an AKS add-on, it is a separate group of resources.

Best example of that is found in the link in the issue.
